Automattic/wp-calypso
# chore(odyssey-stats): upgrade to `react-query` to `v4`

Related to #76077.

## Proposed Changes

* Upgrades `apps/odyssey-stats` to use `@tanstack/react-query@4`

## Testing Instructions

* Open your local Jetpack site.
* Check out the branch of the Jetpack to the latest `trunk`.
* Build Jetpack if necessary: `jetpack build plugins/jetpack`.
* Run `STATS_PACKAGE_PATH=/path/to/jetpack/projects/packages/stats-admin yarn dev` under `calypso/apps/odyssey-stats`.
* Navigate to `/wp-admin/index.php`.
* Ensure that the Jetpack Stats Widget looks and behaves as expected.
